docs/react-tutorial.md

```diff
@@ -0,0 +1,340 @@
+
+```javascript
+function HandleSignIn({ history }) {
+  React.useEffect(() => {
+    transposit.handleSignIn().then(() => {
+      history.replace("/");
+    });
+  }, []);
+  return null;
+}
+```
+
+### _/_ (index)
+
+Require users to sign in before accessing your index page. Use the SDK to create React hooks that implement this behavior.
+
+```javascript
+// Hook to check that user is signed-in. Return true if they are.
+function useSignedIn(history) {
+  const [isSignedIn, setIsSignedIn] = React.useState(false);
+  React.useEffect(() => {
+    if (!transposit.isSignedIn()) {
+      history.push("/signin");
+    } else {
+      setIsSignedIn(true);
+    }
+  }, []);
+  return isSignedIn;
+}
+
+// Hook to load the signed-in user
+function useUser(isSignedIn) {
+  const [user, setUser] = React.useState(null);
+  React.useEffect(() => {
+    if (isSignedIn) {
+      transposit.loadUser().then(u => setUser(u));
+    }
+  }, [isSignedIn]);
+  return user;
+}
+
+function Index({ history }) {
+  // Check if signed-in
+  const isSignedIn = useSignedIn(history);
+  const user = useUser(isSignedIn);
+
+  // If not signed-in, wait while rendering nothing
+  if (!user) {
+    return null;
+  }
+
+  // If signed-in, display the app
+  return (
+    <>
+      <h1>Hello, {user.name}</h1>
+      <button
+        onClick={() => {
+          transposit.signOut(`${window.location.origin}/signin`);
+        }}
+      >
+        Sign Out
+      </button>
+    </>
+  );
+}
+```
+
+## Call your backend
+
+Load dynamic data into your web app. Call your backend when your index page loads.
+
+``` javascript
+function Index({ history }) {
+  // Check if signed-in
+  const isSignedIn = useSignedIn(history);
+  const user = useUser(isSignedIn);
+
+  // Call your backend
+  const [data, setData] = React.useState(null);
+  React.useEffect(() => {
+    if (isSignedIn) {
+      transposit
+        .run("hello") // the name of your deployed operation
+        .then(({ results }) => {
+          setData(results);
+        });
+    }
+  }, [isSignedIn]);
+
+  // If not signed-in, wait while rendering nothing
+  if (!user) {
+    return null;
+  }
+
+  // If signed-in, display the app
+  return (
+    <>
+      <h1>Hello, {user.name}</h1>
+      {data ? (
+        <p>
+          <code>{JSON.stringify(data, undefined, 2)}</code>
+        </p>
+      ) : (
+        <p>loading...</p>
+      )}
+      <button
+        onClick={() => {
+          transposit.signOut(`${window.location.origin}/signin`);
+        }}
+      >
+        Sign Out
+      </button>
+    </>
+  );
+}
+```
+
+Your web app now offers sign-in and calls your backend. The code in its entirety can be viewed [here](./react-tutorial-code.js).
```

Comment:
I'd put this link at the top in case folks don't want the tutorial but just copy/fork instead.